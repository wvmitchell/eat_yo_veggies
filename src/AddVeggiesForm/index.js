import React, { Component } from 'react'

class AddVeggiesForm extends Component {
  constructor(props) {
    super(props)
    this.state = {
      veggie: ''
    }
  }

  updateNewVeggie = (event) => {
    this.setState({ veggie: event.target.value })
  }

  handleSubmit = (event) => {
    event.preventDefault()
    this.props.addVegetable(this.state.veggie)
  }

  render() {
    return(
      <form onSubmit={this.handleSubmit}>
        <input 
          placeholer="New Vegetable" 
          onChange={this.updateNewVeggie} />
        <button type="submit">Add it</button>
      </form>
    )
  }
}

export default AddVeggiesForm
