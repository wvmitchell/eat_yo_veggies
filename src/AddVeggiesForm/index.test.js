import React from 'react'
import { shallow } from 'enzyme'
import AddVeggiesForm from './index'

describe('AddVeggiesForm', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<AddVeggiesForm />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should have a default state', () => {
    const wrapper = shallow(<AddVeggiesForm />)
    expect(wrapper.state()).toEqual({ veggie: '' })
  })

  describe('updateNewVeggie', () => {
    it('should take the change event and update the state', () => {
      // Setup
      const mockEvent = { target: { value: 'squash' } }
      const wrapper = shallow(<AddVeggiesForm />)

      // Execution
      wrapper.instance().updateNewVeggie(mockEvent)

      // Expectation
      expect(wrapper.state('veggie')).toEqual('squash')
    })
  })

  describe('handleSubmit', () => {
    it('should digest the form submission and call addVegetable', () => {
      // Setup
      const mockEvent = { preventDefault: jest.fn() }
      const mockAddVegetable = jest.fn()
      const wrapper = shallow(<AddVeggiesForm addVegetable={mockAddVegetable} />)
      wrapper.setState({ veggie: 'squash' })

      // Execution
      wrapper.instance().handleSubmit(mockEvent)

      // Expectation
      expect(mockEvent.preventDefault).toHaveBeenCalled()
      expect(mockAddVegetable).toHaveBeenCalledWith('squash')
    })
  })
})
