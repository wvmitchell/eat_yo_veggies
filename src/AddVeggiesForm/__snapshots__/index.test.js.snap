// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddVeggiesForm should match the snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    placeholer="New Vegetable"
  />
  <button
    type="submit"
  >
    Add it
  </button>
</form>
`;
